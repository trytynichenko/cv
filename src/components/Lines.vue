<template>
  <div class="lines">
    <div v-for="(item, i) in data" :key="i" class="line">
      <div class="line__row">
        <div class="line__title">{{ item.title }}</div>
        <div class="line__date">{{ item.time }}</div>
      </div>
      <div class="line__description">{{ item.description }}</div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'Lines',
  props: {
    data: Array,
  },
};
</script>

<style scoped lang="scss">
  @import '../assets/styles/components/lines';
</style>
