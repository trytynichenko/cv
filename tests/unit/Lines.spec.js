import { shallow } from '@vue/test-utils';
import Lines from '@/components/Lines.vue';

describe('Lines.vue', () => {
  const props = {
    data: [
      {
        title: 'title 1',
        time: '2018',
        description: 'content 1',
      },
      {
        title: 'title 2',
        time: '2018',
        description: 'content 2',
      },
    ],
  };

  const wrapper = shallow(Lines, {
    propsData: props,
  });

  it('matches snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
