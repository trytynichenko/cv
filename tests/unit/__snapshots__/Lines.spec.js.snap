// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lines.vue matches snapshot 1`] = `
<div class="lines">
    <div class="line">
        <div class="line__row">
            <div class="line__title">title 1</div>
            <div class="line__date">2018</div>
        </div>
        <div class="line__description">content 1</div>
    </div>
    <div class="line">
        <div class="line__row">
            <div class="line__title">title 2</div>
            <div class="line__date">2018</div>
        </div>
        <div class="line__description">content 2</div>
    </div>
</div>
`;
